src/backtrace.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * @packageDocumentation
+ * Collects backtraces for executables for diagnostics
+ */
+import { stringifyError } from "./errors.js";
+import * as actionsCore from "@actions/core";
+import * as exec from "@actions/exec";
+import { readFile, readdir } from "node:fs/promises";
+import { promisify } from "node:util";
+import { gzip } from "node:zlib";
+import os from "os";
+
+export async function collectBacktraces(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  if (os.platform() === "darwin") {
+    return await collectBacktracesMacOS(prefixes);
+  }
+  if (os.platform() === "linux") {
+    return await collectBacktracesSystemd(prefixes);
+  }
+
+  return new Map();
+}
+
+export async function collectBacktracesMacOS(
+  prefixes: string[],
+): Promise<Map<string, string>> {
+  const backtraces: Map<string, string> = new Map();
+
+  try {
+    const { stdout: logJson } = await exec.getExecOutput(
+      "log",
+      [
+        "show",
+        "--style",
+        "json",
+        "--last",
+        // Note we collect the last 1m only, because it should only take a few seconds to write the crash log.
+        // Therefor, any crashes before this 1m should be long done by now.
+        "1m",
+        "--no-info",
+        "--predicate",
+        "sender = 'ReportCrash'",
+      ],
+      {
+        silent: true,
+      },
+    );
+
+    const sussyArray: unknown = JSON.parse(logJson);
+    if (!Array.isArray(sussyArray)) {
+      throw new Error(`Log json isn't an array: ${logJson}`);
+    }
+
+    if (sussyArray.length > 0) {
+      actionsCore.info(`Collecting crash data...`);
+      const delay = async (ms: number): Promise<void> =>
+        new Promise((resolve) => setTimeout(resolve, ms));
+      await delay(5000);
+    }
+  } catch (e: unknown) {
+    actionsCore.debug(
+      "Failed to check logs for in-progress crash dumps, assuming there are none.",
+    );
+  }
+
+  const dirs = [
+    ["system", "/Library/Logs/DiagnosticReports/"],
+    ["user", `${process.env["HOME"]}/Library/Logs/DiagnosticReports/`],
+  ];
+
+  for (const [source, dir] of dirs) {
+    const fileNames = (await readdir(dir)).filter((fileName) => {
+      return prefixes.some((prefix) => fileName.startsWith(prefix));
+    });
+
+    const doGzip = promisify(gzip);
+    for (const fileName of fileNames) {
+      try {
+        const logText = await readFile(`${dir}/${fileName}`);
+        const buf = await doGzip(logText);
+        backtraces.set(
+          `backtrace_value_${source}_${fileName}`,
+          buf.toString("base64"),
+        );
+      } catch (innerError: unknown) {
+        backtraces.set(
+          `backtrace_failure_${source}_${fileName}`,
+          stringifyError(innerError),
+        );
+      }
+    }
+  }
+
+  return backtraces;
+}
+
+interface SystemdCoreDumpInfo {
```

Comment:
Use a `type` here instead. They're kinda the same thing but we generally use `type` when we're defining a straightfoward record type.